storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -607,6 +623,25 @@ private void updateFinalizedData(
               .flatMap(root -> getHotBlock(root).stream())
               .collect(Collectors.toSet()));
     }
+
+    return optimisticTransitionPayload;
+  }
+
+  private Optional<ExecutionPayload> updateFinalizedOptimisticTransitionBlock(
+      final boolean isFinalizedOptimisticBlockRootSet,
+      final Optional<Bytes32> finalizedOptimisticTransitionBlockRoot) {
+    if (isFinalizedOptimisticBlockRootSet) {
```

Comment:
This is the reason why we need an additional boolean aside the Optional, to actually do put\delete only once.
So on DB level we can potentially remove the optimistic transition block slot. Which I guess will be done when the actual validation will be done.

Reply:
Yep exactly - the value itself may be present or absent so we either need a separate boolean to track whether it's set or to use `Optional<Optional<Bytes32>>` and the separate boolean seemed a lot neater and easier to understand.